pkg/deploy/deploy_rp.go

```diff
@@ -90,6 +90,12 @@ func (d *deployer) DeployRP(ctx context.Context) error {
 			continue
 		}
 		if err != nil {
+			// Try to remove failed VMSS so conflict doesn't occur during retry
+			// (when new VMSS with same name is deployed)
+			innerErr := d.rpRemoveFailedScaleset(ctx)
+			if innerErr != nil {
+				d.log.Print(innerErr)
+			}
 			return err
```

Comment:
```suggestion
				return innerErr
			}
			continue
```

And bump the loop (which I cannot seem to put a suggestion on) to be `i < 3` to give this a chance to retry.  There are now 2 error conditions we need to account for so need at least 3 attempts at creating the VMSS.

Reply:
Point taken - I'm not sure that the above suggestion will work exactly, though. We still need to return `err` somewhere. I'll see what I can do.

Reply:
@jewzaam I've updated this in a way I think will work, if you want to take another look